app/models/version.rb

```diff
@@ -23,6 +23,7 @@ class Version < ApplicationRecord # rubocop:disable Metrics/ClassLength
   after_create :create_gem_download
   after_create :record_push_event
   after_save :reorder_versions, if: -> { saved_change_to_indexed? || saved_change_to_id? }
+  after_save :enqueue_web_hook_jobs, if: -> { saved_change_to_indexed? && (!saved_change_to_id? || indexed?) }
```

Comment:
Should this be after_commit so it waits for transactions to finish?

Reply:
it's fine since theyre spooled inside the transaction